### PR TITLE
Add warning logs for manifests, siblings, bytes and history

### PR DESCRIPTION
**Pull request by shino**
Add three kinks of warning logs for potentially bloated manifests.
Original issue is #882.
- many siblings
- large size
- long history ({UUID, M} pairs)

Each has default value and enabled, but has kill switch `disabled`.

### CODE REVIEW

File: src/riak_cs_utils.erl [hunk 1]
```diff
@@ -432,6 +433,40 @@ gc_deleted_while_writing_manifests(Object, Manifests, Bucket, Key, RcPid) ->
     UUIDs = riak_cs_manifest_utils:deleted_while_writing(Manifests),
     riak_cs_gc:gc_specific_manifests(UUIDs, Object, Bucket, Key, RcPid).
 
+-spec maybe_log_warning(binary(), binary(), riakc_obj:riakc_obj(), [term()]) -> ok.
+maybe_log_warning(Bucket, Key, Object, Manifests) ->
```
Comment (kuenishi) [inline src/riak_cs_utils.erl]: Nitpick, the name of this function seems too general. I prefer this be `maybe_warn_manifest_size/4` and in `riak_cs_manifest.erl` .

File: src/riak_cs_utils.erl [hunk 2]
```diff
@@ -432,6 +433,40 @@ gc_deleted_while_writing_manifests(Object, Manifests, Bucket, Key, RcPid) ->
...
+      "Long manifest history for key", "manifests"),
+    ok.
+
+-spec maybe_log_warning(binary(), binary(), disabled | non_neg_integer(),
+                        non_neg_integer(), string(), string()) -> ok.
+maybe_log_warning(Bucket, Key, Actual, Threshold, MessagePrefix, MessageSuffix) ->
+    case Threshold of
+        disabled -> ok;
+        _ when Actual < Threshold -> ok;
+        _ -> _ = lager:warning("~s ~s/~s: ~p ~s",
```
Comment (kuenishi) [inline src/riak_cs_utils.erl]: This `<Bucket>/<Key>` format is confusing to me. How about format like `(<Bucket>, <Key>)`, separated by space?

**Comment by kuenishi** [comment]
I tested overwriting hundreds times, then I got logs like this per single put:

```
2014-07-15 16:41:59.286 [warning] <0.4983.0>@riak_cs_utils:maybe_log_warning:465 Long manifest history for key test/md.xml: 451 manifests
2014-07-15 16:41:59.299 [warning] <0.4985.0>@riak_cs_utils:maybe_log_warning:465 Long manifest history for key test/md.xml: 451 manifests
```

This is maybe because `get_manifets` is called twice somewhere in put_fsm ... I would like to get just once.

**Comment by shino** [comment]
> This is maybe because get_manifets is called twice somewhere in put_fsm ... I would like to get just once.

Both `riak_cs_get_fsm` and `riak_cs_put_fsm` have `riak_cs_manifest_fsm` each.
It may be a sign for refactoring :smile: .
I want to defer it after 1.5 release. Thoughts?

**Comment by shino** [comment]
@kota Addressed your comments except the last one and pushed commit.

Now logs look like these:

```
11:46:55.124 [warning] Long manifest history (32 manifests) for bucket=<<"test2">> key=<<"\t">>
11:48:55.776 [warning] Many manifest siblings (210 siblings) for bucket=<<"test">> key=<<"mp5">>
11:48:55.776 [warning] Large manifest size (7882640 bytes) for bucket=<<"test">> key=<<"mp5">>
```

**Comment by shino** [comment]
Ouch, I mentioned wrong Kota-san. Sorry again and again.

I wanted to mention @kuenishi :sweat:

**Comment by shino** [comment]
@borshop merge

**Pull request merged**